test/common/http/header_map_impl_test.cc

```diff
@@ -907,6 +907,20 @@ TEST(HeaderMapImplTest, Get) {
   }
 }
 
+TEST(HeaderMapImplTest, TestHeaderList) {
+  std::array<Http::LowerCaseString, 2> keys{Headers::get().Path, LowerCaseString("hello")};
+  std::array<std::string, 2> values{"/", "world"};
+
+  auto headers = createHeaderMap<TestHeaderMapImpl>({{keys[0], values[0]}, {keys[1], values[1]}});
+  const auto header_list = headers->createHeaderListView();
+  const auto header_keys = header_list.keys();
+  const auto header_values = header_list.values();
+  for (size_t i = 0; i < keys.size(); ++i) {
```

Comment:
I think there are gtest/gmock expectations that can compare vectors/arrays. Please use those.